src/components/interface/SwitchTheme/SwitchTheme.js

```diff
@@ -2,34 +2,47 @@ import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { ReactComponent as Light } from '@gouvfr/dsfr/dist/artwork/light.svg';
 import { ReactComponent as Dark } from '@gouvfr/dsfr/dist/artwork/dark.svg';
+import { ReactComponent as System } from '@gouvfr/dsfr/dist/artwork/system.svg';
 import { Modal, ModalTitle, ModalContent } from '../Modal';
 import { RadioGroup, Radio } from '../Radio';
-import useTheme from './useTheme';
 
 /*
 * DSFR v1.5.0
 */
 import '@gouvfr/dsfr/dist/scheme/scheme.css';
 
 const SwitchTheme = ({
-  isOpen, setIsOpen, title, legend, darkLabel, lightLabel,
+  isOpen, setIsOpen, title, legend, darkLabel, lightLabel, systemLabel,
 }) => {
-  const currentTheme = useTheme();
+  let initialTheme = window.localStorage.getItem('prefers-color-scheme');
+  if (initialTheme == null) initialTheme = systemLabel;
+  const dark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+  const system = dark ? 'dark' : 'light';
```

Comment:
Why did you remove this code from the useEffect ? 
it does not have to be done every render